src/main/java/com/avispl/symphony/dal/communicator/aver/ptz/AverPTZCommunicator.java

```diff
@@ -0,0 +1,361 @@
+			byte[] expectedSeqNum = convertIntToByteArray(sequenceNumber);
+
+			if (Arrays.equals(expectedSeqNum, responseSeqNum)) {
+				int payloadLength = response[3];
+				byte[] reply = Arrays.copyOfRange(response, 8, 8 + payloadLength);
+				byte currentValue = reply[2];
+
+				if (commandType == CommandType.COMMAND) {
+					if (!Arrays.equals(ReplyStatus.COMPLETION.getCode(), reply)) {
+						if (this.logger.isErrorEnabled()) {
+							this.logger.error("error: Unexpected completion packet: " + this.host + " port: " + this.port);
+						}
+						throw new RuntimeException("Unexpected completion packet");
+					}
+				} else if (commandType == CommandType.INQUIRY) {
+					switch (expectedResponseName) {
+						case "Power": {
+							Optional<PowerStatus> powerStatus = Arrays.stream(PowerStatus.values())
+									.filter(status -> status.getCode() == currentValue)
+									.findFirst();
+
+							return powerStatus.<Object>map(PowerStatus::getName).orElse(null);
+						}
+						case "FocusMode": {
+							Optional<FocusMode> focusMode = Arrays.stream(FocusMode.values())
+									.filter(mode -> mode.getCode() == currentValue)
+									.findFirst();
+
+							return focusMode.<Object>map(FocusMode::getName).orElse(null);
+						}
+						case "AEMode": {
+							Optional<AEMode> aeMode = Arrays.stream(AEMode.values())
+									.filter(mode -> mode.getCode() == currentValue)
+									.findFirst();
+
+							return aeMode.<Object>map(AEMode::getName).orElse(null);
+						}
+						case "AutoSlowShutter": {
+							Optional<SlowShutterStatus> slowShutterStatus = Arrays.stream(SlowShutterStatus.values())
+									.filter(mode -> mode.getCode() == currentValue)
+									.findFirst();
+
+							return slowShutterStatus.<Object>map(SlowShutterStatus::getName).orElse(null);
+						}
+						case "ShutterSpeed":
+						case "IrisLevel":
+						case "GainLevel":
+						case "ExposureValue":
+						case "RGain":
+						case "BGain": {
+							return reply[4] * 16 + reply[5];
+						}
+						case "GainLimitLevel": {
+							return (int) reply[2];
+						}
+						case "Backlight": {
+							Optional<BacklightStatus> backlightStatus = Arrays.stream(BacklightStatus.values())
+									.filter(status -> status.getCode() == currentValue)
+									.findFirst();
+
+							return backlightStatus.<Object>map(BacklightStatus::getName).orElse(null);
+						}
+						case "WBMode": {
+							Optional<WBMode> wbMode = Arrays.stream(WBMode.values())
+									.filter(mode -> mode.getCode() == currentValue)
+									.findFirst();
+
+							return wbMode.<Object>map(WBMode::getName).orElse(null);
+						}
+						case "PanTiltSlowMode": {
+							Optional<SlowPanTiltStatus> slowPanTiltStatus = Arrays.stream(SlowPanTiltStatus.values())
+									.filter(status -> status.getCode() == currentValue)
+									.findFirst();
+
+							return slowPanTiltStatus.<Object>map(SlowPanTiltStatus::getName).orElse(null);
+						}
```

Comment:
I think we missed the "PresetControl" case

Reply:
Oh, sorry, I installed it but I miss add to git :D